### PR TITLE
feat(install, cas templates): adds openebs.io/version label to the OpenEBS dataplane components

### PR DESCRIPTION
**Pull request by mittachaitu**
Signed-off-by: mittachaitu <sai.chaithanya@mayadata.io>

**What this PR does**:
This PR adds the openebs.io/version label to the data plane pods.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #https://github.com/openebs/openebs/issues/2516

**Special notes for your reviewer**:

**Checklist:**
- [x] Fixes #openebs/openebs#2516
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/install/v1alpha1/cstor_pool.go [hunk 1]
```diff
@@ -200,11 +200,13 @@ spec:
       selector:
         matchLabels:
           app: cstor-pool
+          openebs.io/version: {{ .CAST.version }}
```
Comment (vishnuitta) [inline pkg/install/v1alpha1/cstor_pool.go]: why do we need here when version is already part of `metadata.labels`?

Comment (sonasingh46) [reply pkg/install/v1alpha1/cstor_pool.go]: @vishnuitta -- labels present in `matchLabels` are those which should be matched with labels present in metadata for a controller e.g. deployment to manage the pods.
Now we need to see do we also need the version labels as part of this matching.

Comment (mittachaitu) [reply pkg/install/v1alpha1/cstor_pool.go]: Reason to add openebs.io/version labels under `matchLabels` is 1) Deployment controller will first delete the (rs, pods) which doesn't have the matching label in `.spec.template.metadata.labels`(useful in update scenarios). 
2) Deployment controller will never allow to spawn the pods if matching label value is not found in `template.metadata.labels`.

**Comment by vishnuitta** [comment]
setting version need update for upgrade scripts as well.

**Comment by mittachaitu** [comment]
> setting version need update for upgrade scripts as well.

I labeled it with `need-upgrade`

**Comment by mittachaitu** [comment]
> setting version need update for upgrade scripts as well.

I labeled it with `need-upgrade`

**Pull request merged**